hms-api/src/repository/domain/IdeaComment.ts

```diff
@@ -0,0 +1,94 @@
+import Uuid, {uuid} from '../../util/Uuid';
+import ValidationResult from '../../error/ValidationResult';
+
+/**
+ * A Comment represents a single String of text posted by a User on an Idea
+ *
+ * A Comment belongs to 1 and only 1 Idea
+ *
+ * A Comment maps to 1 and only 1 User
+ */
+export default class {
+  /**
+   * The ID of the Comment
+   *
+   * Generated upon creation
+   */
+  id: Uuid;
+
+  /**
+   * The ID of the User which the Comment represents
+   */
+  userId: Uuid;
+
+  /**
+   * The ID of the Idea to which the Comment belongs
+   */
+  ideaId: Uuid;
+
+  /**
+   * the String that contains the text of the Comment
+   */
+  text: string;
+
+  /**
+   * the comment object that this comment is a reply to
+   *
+   * if this is a top level comment, this will be null
+   */
+  replyTo: Uuid;
```

Comment:
this should be nullable
```suggestion
  replyTo: Uuid?;
```

Reply:
<img width="661" alt="Bildschirmfoto 2023-01-31 um 14 08 44" src="https://user-images.githubusercontent.com/59581396/215768412-6d113693-9ab8-4a4b-841d-abedb4626025.png">

Reply:
works with replyTo?: Uuud; but this causes a lot of issues with the order because a optional parameter needs be placed after all needed parameters so have to change a lot und will do this tomorrow

Reply:
ah wait that was kotlin syntax🤦, should be like this:

```suggestion
  replyTo?: Uuid;
```

or even possibly...
```
  replyTo: Uuid | null;
```

but I think the first is better